pycbc/waveform/waveform.py

```diff
@@ -503,6 +503,63 @@ def get_fd_waveform(template=None, **kwargs):
     params=parameters.fd_waveform_params.docstr(prefix="    ",
            include_label=False).lstrip(' '))
 
+def get_td_waveform_from_fd(**params):
+    """ Return time domain version of fourier domain approximant.
+
+    This returns a time domain version of a fourier domain approximant, with
+    padding and tapering at the start of the waveform.
+
+    Parameters
+    ----------
+    params: dict
+        The parameters defining the waveform to generator.
+        See `get_fd_waveform`.
+
+    Returns
+    -------
+    hp: pycbc.types.TimeSeries
+        Plus polarization time series
+    hc: pycbc.types.TimeSeries
+        Cross polarization time series
+    """
+
+    # determine the duration to use
+    full_duration = duration = get_waveform_filter_length_in_time(**params)
+    nparams = params.copy()
+    if '_FD' in nparams['approximant']:
+        nparams['approximant'] = nparams['approximant'][:-3]
+
+    while full_duration < duration * 1.5:
+        full_duration = get_waveform_filter_length_in_time(**nparams)
+        nparams['f_lower'] -= 1
+
+    if 'f_fref' not in nparams:
+        nparams['f_ref'] = params['f_lower']
+
+    # factor to ensure the vectors are all large enough. We don't need to
+    # completely trust our duration estimator in this case, at a small
+    # increase in computational cost
+    rwrap = 0.2
+    fudge_duration = (max(0, full_duration) + .1 + rwrap) * 1.5
+    nparams['delta_f'] = 1.0 / fudge_duration
```

Comment:
I think you need to round `fudge_duration` to the nearest power of two here. Otherwise, the memory usage might grow a bunch if you used this multiple times with different parameters, due to the FFT-plan caching that silently happens in `to_timeseries`.